instrumentation/helpers-sql-obfuscation/lib/opentelemetry/instrumentation/helpers/sql_obfuscation.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0module OpenTelemetry
+
+module OpenTelemetry
+  module Instrumentation
+    module Helpers
+      #
+      # This class contains SQL obfuscation behavior to share with
+      # instrumentation for specific database adapters.
+      # The class uses code from: https://github.com/newrelic/newrelic-ruby-agent/blob/1fca78cc7a087421ad58088d8bea72c0362bc62f/lib/new_relic/agent/database/obfuscation_helpers.rb
+      #
+      # To use this in your instrumentation, the `Instrumentation` class for
+      # your gem must contain configuration options for:
+      #  * `:db_statement`
+      #    Example:
+      #    `option :db_statement, default: :include, validate: %I[omit include obfuscate]`
+      #  * `:obfuscation_limit`
+      #    Example:
+      #    `option :obfuscation_limit, default: 2000, validate: :integer`
+      #
+      # If you want to add support for a new adapter, update the following
+      # constants to include keys for your adapter:
+      #  * DIALECT_COMPONENTS
+      #  * CLEANUP_REGEX
+      # You must also add a new constant that uses `generate_regex` with your
+      # adapter's dialect components that is named like
+      # `<ADAPTER>_COMPONENTS_REGEX`, such as: `MYSQL_COMPONENTS_REGEX`.
+      #
+      # @api public
+      class SqlObfuscation
+        # From: https://github.com/newrelic/newrelic-ruby-agent/blob/1fca78cc7a087421ad58088d8bea72c0362bc62f/lib/new_relic/agent/database/obfuscation_helpers.rb
+        COMPONENTS_REGEX_MAP = {
+          single_quotes: /'(?:[^']|'')*?(?:\\'.*|'(?!'))/,
+          double_quotes: /"(?:[^"]|"")*?(?:\\".*|"(?!"))/,
+          dollar_quotes: /(\$(?!\d)[^$]*?\$).*?(?:\1|$)/,
+          uuids: /\{?(?:[0-9a-fA-F]\-*){32}\}?/,
+          numeric_literals: /-?\b(?:[0-9]+\.)?[0-9]+([eE][+-]?[0-9]+)?\b/,
+          boolean_literals: /\b(?:true|false|null)\b/i,
+          hexadecimal_literals: /0x[0-9a-fA-F]+/,
+          comments: /(?:#|--).*?(?=\r|\n|$)/i,
+          multi_line_comments: %r{\/\*(?:[^\/]|\/[^*])*?(?:\*\/|\/\*.*)}
+        }.freeze
+
+        DIALECT_COMPONENTS = {
```

Comment:
the more i look at this code, the more i continue to think this represents what really ought to be a Specification effort to standardize OpenTelemetry's treatment of "sql obfuscation" as a feature. Common dialect component keys ought to be part of otel semconv, and the implementation of how to accomplish that obfuscation could vary under the hood (regex might not be the most performant approach in some languages, etc), but it would be possible to do a matrix-like comparison of language-sdk's + collector's "obfuscation abilities" on sql(or other _things_, mongodb redis etc), and the varying concerns wrt PII related compliance things that different end users of different language sdks should need to worry about, and possibly, later on in their telemetry pipeline, apply some additional redaction or more fine grained ingestion or indexing approach to their data.

not blocking or anything, just was thinking about this and wanted to socialize it.